OHIF/Viewers
# feat(SidePanel): Left or right side panel to auto start as closed but auto open when needed

### Context

OHIF issue #3135



### Changes & Results

- Added a method to set a callback for a Panel to invoke when it is ready to be shown (e.g. its data is loaded).
- Implemented such methods for both the segmentation and measurement panels.
- The SidePanel component now adds a callback to Panel components so that it will automatically open a Panel that was initially closed and yet to be opened.
- Updated the OHIF documentation accordingly.

### Testing

#### Test A
1. Launch OHIF - the right side panel should be closed/collapsed.
2. Add a measurement.
3. The right side panel should open.

#### Test B
1. Launch OHIF with a study with an SR series that tracked measurement - the right side panel should be closed/collapsed initially.
2. Drag and drop or double click the SR series.
3. Load the measurements.
4. The right side panel should open.

#### Test C
1. Launch OHIF with a study with an SR series that tracked measurement - the right side panel should be closed/collapsed initially.
2. Open and close the right side panel.
3. Drag and drop or double click the SR series.
4. Load the measurements.
5. The right side panel should remain closed.

#### Test D
1. Launch OHIF with a study with a segmentation series - the right side panel should be closed/collapsed initially.
3. Drag and drop or double click the segmentation series.
4. The right side panel should open.

### Checklist

#### PR



  semantic-release format and guidelines.

#### Code

  etc.)

#### Public Documentation Updates



  additions or removals.

#### Tested Environment

- [x] "Browser: Chrome 110.0.5481.178"
  


[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch
